### ISSUE TITLE
Improve Testing on Windows OS

### ISSUE THREAD

**Issue by baolsen** [apache/airflow#12874]
**Description**

Airflow does not currently support Windows (although there is some ability to run on WSL2).
This not only applies to installing / running Airflow but also extends to developing Airflow itself.

**Use case / motivation**

I develop on Windows (corporate requirement) and would like to improve the ability to test Airflow more easily.
My goal isn't to get the full test environment working on Windows, but just take some small steps to get to a point where it is somewhat functional. Using WSL is not ideal for me specifically as I can't install it on my development machine.

**Related Issues**

There is a lot of discussion already here: https://github.com/apache/airflow/issues/10388
Basically, getting Airflow on Windows is a large undertaking. This feature will assist with a (very) small part of it :)

**Comment by baolsen** [comment]
Please assign to me :)

**Comment by potiuk** [comment]
Cool. Happy to help with that! I have a dual boot windows on my PC, so maybe after 2.0 is out and we Rewrite Breeze in Python #12282  that might be actually doable!

**Comment by potiuk** [comment]
I will make sure to add Windows support when I finally have time to finish the design doc :). @baolsen -> please make sure to follow the  discussion and maybe you want to join the team of future Breeze developers :).

**Comment by eladkal** [comment]
I think this issue is covered by https://github.com/apache/airflow/issues/19966
@potiuk can you confirm?

**Comment by potiuk** [comment]
It's certainly part of it. This issue will be fixed only when we havemore or less working breeze on Windows. 

BTW. @baolsen -> we have been struggling a bit with @bowrna and @edithturn with making autocomplete works for Windows. We are using Python's [Click](https://click.palletsprojects.com/en/8.0.x/) for command line parameters and it generates autocomplete Power Shell scripts, but the documentation on it paired wtith our (in)experience with Windows makes it rather pathetic (I am talking about myself :D). Maybe you would like to be one of our early users to test things like that on Windows? 

I think we need someone like you who uses Windows and knows Airflow and would like to have nice development envronment for it. 

Would it be possible that you help us with it ?

**Comment by potiuk** [comment]
@baolsen We are unassigning you from this issue as part of our updated [assignment policy](https://github.com/apache/airflow/blob/main/contributing-docs/04_how_to_contribute.rst#contribute-code-changes).

This is not meant to discourage your contribution — quite the opposite! You are still very welcome to work on this issue and submit a PR for it. Simply comment that you are working on it and open a PR when ready.

We found that formal assignments were not working well, as they often prevented others from contributing when the assignee was not actively working on the issue.
